src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_is_handled_by_slr.cs

```diff
@@ -0,0 +1,165 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_message_is_handled_by_slr
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_not_send_outgoing_messages(TransportTransactionMode transactionMode)
+        {
+            await Scenario.Define<Context>(c =>
+            {
+                c.Id = Guid.NewGuid();
+                c.TransactionMode = transactionMode;
+            })
+            .WithEndpoint<Endpoint>(b => b.DoNotFailOnErrorMessages()
+                .When((session, context) => session.SendLocal(new InitiatingMessage
+                {
+                    Id = context.Id
+                }))
+            )
+            .WithEndpoint<ReceiverEndpoint>()
+            .WithEndpoint<ErrorSpy>()
+            .Done(c => c.MessageMovedToErrorQueue)
```

Comment:
I think that we need to merge the Receiver and ErrorSpy and set ErrorSpy concurrency limit to 1. Otherwise it is possible that message will be moved to error queue and processed before the receiver gets it.